server/src/main/java/org/opensearch/common/time/JavaDateFormatter.java

```diff
@@ -101,14 +101,18 @@ JavaDateFormatter getRoundupParser() {
     JavaDateFormatter(
         String format,
         String printFormat,
-        DateTimeFormatter printer,
+        OpenSearchDateTimeFormatter printer,
```

Comment:
Why do we need to wrap printer? I think we could keep it as DateTimeFormatter

Reply:
I think we should keep printer as `OpenSearchDateTimeFormatter` because a hand written parser can provide print capabilities as well. Although this one doesn't have print support but the design would be extensible

Reply:
Fair enough, thanks. Since we have the opportunity to introduce the dedicated abstraction, let is introduce `OpenSearchDateTimePrinter` (which OpenSearchDateTimeFormatter could implement) - it needs much smaller API footprint.

Reply:
Sure, I've introduced the `OpenSearchDateTimePrinter` interface. Let me know if this looks good